### PR TITLE
Move product detail GraphQl fields to fragment

### PR DESCRIPTION
**Pull request by huykon**
## Description

Base on [#1027](https://github.com/magento/graphql-ce/issues/1027) Use ProductInterface fragment for product detail query.

### Verification Steps

1. Go to the product detail page.
2. Verify the product data shows up.
3. Make sure everything working well.

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/RootComponents/Product/product.gql.js [hunk 1]
```diff
@@ -1,92 +1,15 @@
...
 
+import { ProductDetailsFragment } from './productDetailFragment.gql';
+
 export const GET_PRODUCT_DETAIL_QUERY = gql`
     query getProductDetailForProductPage($urlKey: String!) {
         products(filter: { url_key: { eq: $urlKey } }) {
             items {
-                # Once graphql-ce/1027 is resolved, use a ProductDetails fragment
-                # here instead.
-                __typename
```
Comment (sirugh) [inline packages/venia-ui/lib/RootComponents/Product/product.gql.js:9]: I believe `__typename` got lost in the copy to the fragment. Can you add it back?

File: packages/venia-ui/lib/components/CartPage/ProductListing/EditModal/productForm.js [hunk 2]
```diff
@@ -120,49 +121,11 @@ export const GET_CONFIGURABLE_OPTIONS = gql`
...
-                                        currency
-                                        value
-                                    }
-                                }
-                            }
-                            sku
-                        }
-                    }
-                }
+                ...ProductDetailsFragment
```
Comment (sirugh) [inline packages/venia-ui/lib/components/CartPage/ProductListing/EditModal/productForm.js:124]: Unfortunately you cannot just re-use the fragment here and the content needed by the `RootComponent/Product` and the content needed by this component are different. Re-using the fragment will result in over-fetching of data. 

Please revert this change, or create a `productFormFragment`.

File: packages/venia-ui/lib/components/LegacyMiniCart/editItem.js [hunk 3]
```diff
@@ -55,51 +56,10 @@ export const PRODUCT_DETAILS = gql`
...
-                                file
-                                label
-                                position
-                            }
-                            sku
-                            stock_status
-                        }
-                    }
-                }
+                ...ProductDetailsFragment
```
Comment (sirugh) [inline packages/venia-ui/lib/components/LegacyMiniCart/editItem.js:60]: Same comment here. Unfortunately you cannot just re-use the fragment here and the content needed by the `RootComponent/Product` and the content needed by this component are different. Re-using the fragment will result in over-fetching of data. 

Please revert this change, or create an `editFormFragment`.

**Review by sirugh — CHANGES_REQUESTED** [review]
I checked the fragment use and it does seem like graphql-ce/1027 was resolved and the fragment returns the correct data. I would still want to do a regression on the PDP.

While you're here making changes to the GQL files, would you mind moving the Product RootComponent query and fragment to peregrine? You can import and use like the [`useBreadcrumbs` talon](https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/talons/Breadcrumbs/useBreadcrumbs.js#L37).

**Review by sirugh — APPROVED** [review]
I think I see why you didn't move the other fragments to talons as those components define their gql in the same file. I think that is fine for now, so please don't feel like it's necessary to do any further.

Thanks for this!

**Comment by huykon** [comment]
Thank you! I hope this done :)

**Comment by dpatil-magento** [comment]
QA Approved.

**Pull request merged**